app/src/middleware/authorization.js

```diff
@@ -5,7 +5,46 @@ const log = require('../components/log')(module.filename);
 const { AuthMode, AuthType, Permissions } = require('../components/constants');
 const { getAppAuthMode, getCurrentIdentity } = require('../components/utils');
 const { NIL: SYSTEM_USER } = require('uuid');
-const { objectService, permissionService, userService } = require('../services');
+const { bucketPermissionService, objectService, objectPermissionService, userService } = require('../services');
+
+/**
+ * @function _checkPermission
+ * Checks if the current user is authorized to perform the operation
+ * @param {object} req.currentObject Express request object currentObject
+ * @param {object} req.currentUser Express request object currentUser
+ * @param {object} req.params Express request object params
+ * @param {string} permission The permission to check against
+ * @returns {boolean} True if permission exists; false otherwise
+ */
+const _checkPermission = async ({ currentObject, currentUser, params }, permission) => {
+  const authType = currentUser ? currentUser.authType : undefined;
+  let result = false;
+
+  // Guard against unauthorized access for all other cases
+  const userId = await userService.getCurrentUserId(getCurrentIdentity(currentUser, SYSTEM_USER));
+
+  if (authType === AuthType.BEARER && userId) {
+    const permissions = [];
+    const searchParams = { permCode: permission, userId: userId };
+
+    if (params.objId) {
+      permissions.push(...await objectPermissionService.searchPermissions({
+        objId: params.objId, ...searchParams
+      }));
+    }
+    if (params.bucketId || currentObject.bucketId) {
```

Comment:
not quite sure why you're checking currentObject.bucketId

Reply:
Since _checkPermission needs to do discovery and dynamic behavior for both /bucket and /object endpoints, the bucketId may come from multiple places. For Objects, you may have the bucketId specified in currentObject. For bucket, you will have bucketId coming in as a param.